apps/rush-lib/src/cli/actions/ChangeAction.ts

```diff
@@ -100,32 +108,192 @@ export class ChangeAction extends BaseRushAction {
         'determine which projects were changed. If this parameter is not specified, the checked out branch ' +
         'is compared against the "master" branch.'
     });
+
+    this._overwriteFlagParemter = this.defineFlagParameter({
+      parameterLongName: '--overwrite',
+      description: `If a changefile already exists, overwrite without prompting ` +
+        `(or erroring in ${BULK_LONG_NAME} mode).`
+    });
+
+    this._changeEmailParameter = this.defineStringParameter({
+      parameterLongName: '--email',
+      argumentName: 'EMAIL',
+      description: 'The email address to use in changefiles. If this parameter is not provided, the email address ' +
+        'will be detected or prompted for in intractive mode.'
+    });
+
+    this._bulkChangeParameter = this.defineFlagParameter({
+      parameterLongName: BULK_LONG_NAME,
+      description: 'If this flag is specified, apply the same change message and bump type to all changed projects. ' +
+        `The ${BULK_MESSAGE_LONG_NAME} and the ${BULK_BUMP_TYPE_LONG_NAME} parameters must be specified if the ` +
+        `${BULK_LONG_NAME} parameter is specified`
+    });
+
+    this._bulkChangeMessageParameter = this.defineStringParameter({
+      parameterLongName: BULK_MESSAGE_LONG_NAME,
+      argumentName: 'MESSAGE',
+      description: `The message to apply to all changed projects if the ${BULK_LONG_NAME} flag is provided.`
+    });
+
+    this._bulkChangeBumpTypeParameter = this.defineChoiceParameter({
+      parameterLongName: BULK_BUMP_TYPE_LONG_NAME,
+      alternatives: [...Object.keys(this._getBumpOptions()), ChangeType[ChangeType.none]],
+      description: `The bump type to apply to all changed projects if the ${BULK_LONG_NAME} flag is provided.`
+    });
   }
 
   public run(): Promise<void> {
     console.log(`The target branch is ${this._targetBranch}`);
     this._projectHostMap = this._generateHostMap();
 
     if (this._verifyParameter.value) {
+      const errors: string[] = ([
+        this._bulkChangeParameter,
+        this._bulkChangeMessageParameter,
+        this._bulkChangeBumpTypeParameter,
+        this._overwriteFlagParemter
+      ]).map((parameter) => {
+        return parameter.value
+        ? (
+          `The {${this._bulkChangeParameter.longName} parameter cannot be provided with the ` +
+            `${this._verifyParameter.longName} parameter`
+          )
+        : '';
+      }).filter((error) => error !== '');
+      if (errors.length > 0) {
+        errors.forEach((error) => console.error(error));
+        throw new AlreadyReportedError();
```

Comment:
The synchronous preamble for this `Promise<void>` seems to comprise most of the function.  In each place where an exception is thrown, it will be synchronous throw and not a `Promise` rejection.  I know that Rush has protective wrappers for this mistake in certain places, so maybe it works correctly.

But properly speaking, this whole function should be inside a `return Promise.resolve().then(() => { .. .})` wrapper that captures any exceptions.